packages/workspace/src/workspace/element_selector.ts

```diff
@@ -142,118 +123,93 @@ export const createElementSelectors = (selectors: string[]):
-  return compact ? filterOutChildSelectors(selectors, currentLevelElementsSelected) : selectors
-}
-
-const isTopLevelSelector = (selector: ElementSelector, depth: number): boolean => {
-  if (ElemIDTopLevelTypes.includes(selector.idTypeSelector)) {
-    return (!selector.nameSelectors || selector.nameSelectors.length <= depth)
-  }
-  return (!selector.nameSelectors || (selector.nameSelectors.length <= depth - 1))
-}
-
-const getPossiblePathsFromParent = (parent: Element): string[] => {
-  const paths = ['annotations']
-  if (isType(parent)) {
-    paths.push('annotationTypes')
-  }
-  if (isObjectType(parent)) {
-    paths.push('fields')
+const createTopLevelSelector = (selector: ElementSelector): ElementSelector => {
+  if (ElemID.TOP_LEVEL_ID_TYPES_WITH_NAME.includes(selector.idTypeSelector)) {
+    return {
+      adapterSelector: selector.adapterSelector,
+      idTypeSelector: selector.idTypeSelector,
+      typeNameSelector: selector.typeNameSelector,
+      nameSelectors: selector.nameSelectors?.slice(0, 1),
+      origin: selector.origin.split(ElemID.NAMESPACE_SEPARATOR).slice(0,
+        ElemID.NUM_ELEM_ID_NON_NAME_PARTS + 1).join(ElemID.NAMESPACE_SEPARATOR),
+    }
   }
-  if (isInstanceElement(parent)) {
-    paths.push('value')
+  const idType = ElemID.TOP_LEVEL_ID_TYPES.includes(selector.idTypeSelector) ? selector.idTypeSelector : 'type'
+  return {
+    adapterSelector: selector.adapterSelector,
+    idTypeSelector: idType,
+    typeNameSelector: selector.typeNameSelector,
+    origin: [selector.adapterSelector.source, selector
+      .typeNameSelector.source, idType].join(ElemID.NAMESPACE_SEPARATOR),
   }
-  return paths
 }
 
-const pathNameToIdType = (path: string): string => ((path === 'annotations') ? 'annotation.' : '')
+const isTopLevelSelector = (selector: ElementSelector): boolean => ElemID
+  .TOP_LEVEL_ID_TYPES.includes(selector.idTypeSelector)
+  || (ElemID.TOP_LEVEL_ID_TYPES_WITH_NAME.includes(selector.idTypeSelector)
+  && selector.nameSelectors?.length === 1)
 
-const isNestedElement = (parentElement: ElementIDToValue): boolean =>
-  parentElement.elemID.getFullNameParts().length > ElemID.NUM_ELEM_ID_NON_NAME_PARTS
+const createSameDepthSelector = (selector: ElementSelector, elemID: ElemID): ElementSelector =>
+  createElementSelector(selector.origin.split(ElemID.NAMESPACE_SEPARATOR).slice(0,
+    elemID.getFullNameParts().length).join(ElemID.NAMESPACE_SEPARATOR))
 
-const createPathToPropertyMapping = (parentElement: ElementIDToValue, path: string):
-[string, unknown][] => ((parentElement.element[path]) ? Object.entries(parentElement.element[path])
-  .map((entry): [string, unknown] => [`${pathNameToIdType(path)}${entry[0]}`, entry[1]]) : [])
+const isElementPossiblyParentOfSearchedElement = (selectors: ElementSelector[],
+  testId: ElemID): boolean => !(_.isEmpty(selectElementsBySelectors([testId],
+  selectors.map(selector => createSameDepthSelector(selector, testId)), false)))
```

Comment:
can we format this more consistently with the common code style?
I am writing it here just because I had a hard time reading the code as it is formatted now but I think the formatting will help future readers as well
```suggestion
const isElementPossiblyParentOfSearchedElement = (
  selectors: ElementSelector[], testId: ElemID
): boolean => !_.isEmpty(
  selectElementsBySelectors(
    [testId],
    selectors.map(selector => createSameDepthSelector(selector, testId)),
    false
  )
)
```

after reading this re-formatted, I think there is an issue here (unless I made a mistake when re-formatting)
this will run `!_.isEmpty` on an object that contains `elements` and `matches` and therefore it will always return `true`.

I wonder why we need all this complexity though when we are testing one specific element ID (this repeats multiple times in the new code).
wouldn't it be much easier to read and understand if it was
```suggestion
const isElementPossiblyParentOfSearchedElement = (
  selectors: ElementSelector[],
  testId: ElemID,
): boolean => (
  selectors.some(selector => match(testId, createSameDepthSelector(selector, testId))
)
```